platforms/web/scripts/i18next/parse-translations.ts

```diff
@@ -0,0 +1,66 @@
+import { existsSync, readdirSync, writeFileSync } from 'node:fs';
+import { join } from 'node:path';
+import { execSync } from 'node:child_process';
+
+import prettier from 'prettier';
+
+const i18nextParserConfig = join(process.cwd(), 'i18next-parser.config.js');
+const platformGlob = join(process.cwd(), 'src/**/*.{ts,tsx}');
+
+type WorkspaceEntry = {
+  location: string;
+  workspaceDependencies: string[];
+};
+
+type WorkspaceInfo = Record<string, WorkspaceEntry>;
+
+function syncTranslations() {
+  const result = execSync('yarn workspaces --json info').toString('utf-8');
+  const info = JSON.parse(result);
+  const data = JSON.parse(info.data) as WorkspaceInfo;
+
+  const dependencies = data['@jwp/ott-web']?.workspaceDependencies || [];
+
+  const globs = dependencies
+    .map((dependency) => data[dependency])
+    .filter((entry) => entry && existsSync(join('../..', entry.location, 'src')))
+    .map((entry) => `'${join(entry.location, 'src/**/*.{ts,tsx}')}'`)
+    .join(' ');
+
+  const command = `npx i18next --config ${i18nextParserConfig} '${platformGlob}' ${globs}`;
+
+  console.info('Running command:');
+  console.info(` ${command}`);
+
+  execSync(command, { env: process.env, stdio: 'pipe', cwd: '../../' });
```

Comment:
This is half nitpicking and half future proofing, but maybe it would be prudent to calculate the workspace root? If we assume that we have the root of the jwp/web package, we can simply remove the location that we got from workspace info. Or we can use something like this: https://www.npmjs.com/package/find-yarn-workspace-root

Reply:
Yes, good point. CodeQL is also warning about this uncontrolled path.